packages/eslint-plugin/rules/jsx-a11y-anchor-has-content.js

```diff
@@ -0,0 +1,161 @@
+ * elementType from jsx-ast-utils: to get the type of the node we are working with.
+ */
+const anchorRule = require( 'eslint-plugin-jsx-a11y' ).rules[
+	'anchor-has-content'
+];
+/**
+ * Internal dependencies
+ */
+const {
+	getTextContentFromNode,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+} = require( '../utils' );
+
+const meta = {
+	type: 'problem',
+	docs: {
+		description:
+			'Checks that anchors are not empty and have an href, relies on eslint-plugin-jsx-a11y/anchor-has-content under the hood. Supports usage of createInterpolateElement.',
+	},
+	fixable: 'code',
+	messages: {
+		anchorHasContent:
+			'Anchors must have content and the content must be accessible by a screen reader. Check the first parameter to createInterpolateElement.',
+		invalidMarkup:
+			'The first parameter to createInterpolateElement does not contain valid markup for an anchor.',
+	},
+	schema: [
+		{
+			type: 'object',
+			properties: {
+				components: {
+					type: 'array',
+					items: {
+						type: 'string',
+					},
+					uniqueItems: true,
+					additionalItems: false,
+				},
+			},
+		},
+	],
+};
+
+/**
+ *
+ * @param {*} tags     markup used in createInterpolateElement
+ * @param {*} tagNames names that we recognize as anchors, as configured in the eslint rules
+ * @return {Array} tagNames an array of the content of the <a> (or other) tags
+ */
+const getATagsContent = ( tags, tagNames ) => {
+	const tagText = tagNames.join( '|' );
+	const regex = new RegExp( `<(${ tagText })>(.*?)</(${ tagText })>`, 'g' );
+	const tagParts = [ ...tags.matchAll( regex ) ];
+	return tagParts.map( ( element ) => {
+		return element[ 2 ].trim();
+	} );
+};
+
+/**
+ *
+ * @param {Node} node the first param of createInterpolateElement
+ * @return {string|null} a string of the content of the node
+ */
+const getValueFromInterpolateElement = ( node ) => {
+	let nodeStr = null;
+	let args = [];
+	if ( node.type === 'CallExpression' ) {
+		const argFunctionName = getTranslateFunctionName( node.callee );
+		// get text if we are going through a translate function
+		args = getTranslateFunctionArgs(
+			argFunctionName,
+			node.arguments,
+			false
+		).map( getTextContentFromNode );
+		// An unknown function call in a translate function may produce a valid string
+		// but verifying it is not straight-forward, so we bail
+		if ( args.filter( Boolean ).length === 0 ) {
+			nodeStr = null;
+		}
+		nodeStr = args.join( '' );
+	} else if ( node.type === 'Literal' ) {
+		nodeStr = getTextContentFromNode( node );
+	}
+	return nodeStr;
+};
+
+const rule = function ( context ) {
+	return {
+		JSXOpeningElement: ( node ) => {
+			/**
+			 * Uses the custom components rules to determine if the node is an anchor or not.
+			 */
+			const options = context.options[ 0 ] || {};
+			const componentOptions = options.components || [];
+			const typeCheck = [ 'a' ].concat( componentOptions );
+			if (
+				typeCheck.filter( ( type ) => type === node?.name?.name )
```

Comment:
Do we also need to filter out nullish values before we check the `length`?

Reply:
I didn't think so, but I've added a check to be safe in case something is misconfigured, thanks for catching it